### PR TITLE
Preference page

### PR DESCRIPTION
**Pull request by cknguyen123**

### CODE REVIEW

**Comment by LoukikNaik** [comment]
The bio editText field gets hidden behind the keyboard when the keyboard pops up.
Wrapping constrained layout in `fragment_preferences.xml` with `ScrollView` seems to solve the problem.

**Comment by LoukikNaik** [comment]
`<ScrollView
    xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:fillViewport="true"
    android:layout_width="match_parent"
    android:layout_height="match_parent">`

File: app/src/main/java/com/example/activeamigo/PreferenceActivity.java [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+
+    protected EditText dobEditText;
+    protected Button edit_save;
+    protected Spinner exercise_choice;
+    protected Spinner location_choice;
+    protected RadioGroup genderGroup;
+    protected String gender;
+    protected EditText bio;
+
+    protected String collection="Accounts";
```
Comment (amquach00) [inline app/src/main/java/com/example/activeamigo/PreferenceActivity.java]: please make static

File: app/src/main/java/com/example/activeamigo/PreferenceActivity.java [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+    protected String document = "test@ucsd.edu";
```
Comment (amquach00) [inline app/src/main/java/com/example/activeamigo/PreferenceActivity.java]: please make static

File: app/src/main/java/com/example/activeamigo/PreferenceActivity.java [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+public class PreferenceActivity extends AppCompatActivity {
+
+    protected EditText dobEditText;
```
Comment (amquach00) [inline app/src/main/java/com/example/activeamigo/PreferenceActivity.java]: please make variables on line 37-43 private

File: app/src/main/java/com/example/activeamigo/PreferenceActivity.java [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+        if (bio != null) {
+            bio.setText(bioText);
+        }
+    }
+
+    /** Function to check if all fields are inputted. Returns true if everything is inputted or else it will reutrn false and prompt the user**/
+    private boolean checkFieldsCompleted(){
+        if (exercise_choice.getSelectedItemPosition() == 0 || location_choice.getSelectedItemPosition() == 0) {
+            // Spinner is at index 0 (first item) indicating "Select Activity" or "Select Location" is still selected
+            Toast.makeText(this, "Please select valid options for activity and location", Toast.LENGTH_SHORT).show();
```
Comment (amquach00) [inline app/src/main/java/com/example/activeamigo/PreferenceActivity.java]: add these strings in the strings.xml file

**Review by amquach00 — DISMISSED** [review]
everything lgtm

File: app/src/main/java/com/example/activeamigo/PreferenceActivity.java (hunk 1 above)
Comment (cknguyen123) [reply app/src/main/java/com/example/activeamigo/PreferenceActivity.java]: needs to be private and modifiable. cannot be static

**Pull request merged**